browser/brave_content_browser_client.cc

```diff
@@ -253,11 +253,9 @@ bool BraveContentBrowserClient::HandleExternalProtocol(
 
 base::Optional<service_manager::Manifest>
 BraveContentBrowserClient::GetServiceManifestOverlay(base::StringPiece name) {
-  auto manifest = ChromeContentBrowserClient::GetServiceManifestOverlay(name);
-  if (name == content::mojom::kBrowserServiceName) {
-    manifest->Amend(GetBraveContentBrowserOverlayManifest());
-  }
-  return manifest;
+  if (name == content::mojom::kBrowserServiceName)
+    return GetBraveContentBrowserOverlayManifest();
```

Comment:
is this something we should look at changing as a follow-up?

Reply:
cc: @mariospr

Reply:
Relying on service manifests for now is ok since there are still a few internal services upstream that rely on it, but it would certainly be good to start thinking of migrating to another mechanism that doesn't rely on them for the `bat_ledger`, `bat_ads` and `tor_launcher` Brave-specific services, in preparation for the future.

I imagine that something along the lines of what BrowserInterfaceBroker (BIB) does for renderer->browser-process communication (but for browser->Brave-specific-service instead) could be the way to go, but I'd need to double check a few things that to be fully sure. I'll take a look to that in parallel to the cr87 rebase work and will file an issue in github once I gather all the info I need, if that's ok.

For this particular rebase, though, I think this is the way to go since it keeps the original behaviour with minimal changes. 

@bridiver WDYT?

Reply:
@mariospr Is this the same as migrating services off service manager? I believe we have plans to do so for these services, we created an issue for TorLauncher https://github.com/brave/brave-browser/issues/8761, I was working on that but got interrupted, I could get back to it and also make the same changes to rewards and ads too.

Reply:
> @mariospr Is this the same as migrating services off service manager? I believe we have plans to do so for these services, we created an issue for TorLauncher brave/brave-browser#8761 [...]

@yrliou It is exactly the same thing, yes! I was about to link crbug.com/977637 and a couple of example migrations from upstream (e.g. crrev.com/c/1905050, crrev.com/c/1963498) that could be relevant, but it turns out that [the PR associated to the brave-browser issue (8761) that you linked](https://github.com/brave/brave-core/pull/6623/files#diff-9407668151e064f23ff5fb73acd19efc) is exactly what needs to be done!

>  [...] I was working on that but got interrupted, I could get back to it and also make the same changes to rewards and ads too.

From what I can see, the only services that would need to be moved off the ServiceManager in the context of this discussion are `bat_ledger`, `bat_ads` and `tor_launcher`, and for that it seems that https://github.com/brave/brave-core/pull/6623 is already tackling those. Either way, it would be great to have that PR merged so that we can drop this dependency on service manifests, which will eventually go away upstream.